src/overflow/setup_bands.py

```diff
@@ -0,0 +1,32 @@
+from osgeo import gdal
+from contextlib import contextmanager
+
+
+gdal.UseExceptions()
+
+
+@contextmanager
+def setup_bands(input_path, output_path):
+    n_bands = 1
+    input_raster = gdal.Open(input_path)
+    driver = gdal.GetDriverByName("GTiff")
+    output_dataset = driver.Create(
+        output_path,
+        input_raster.RasterXSize,
+        input_raster.RasterYSize,
+        n_bands,
+        gdal.GDT_Float32,
+    )
+
+    projection = input_raster.GetProjection()
+    output_dataset.SetProjection(projection)
+    transform = input_raster.GetGeoTransform()
+    output_dataset.SetGeoTransform(transform)
+
+    band_id = 1
+    input_band = input_raster.GetRasterBand(band_id)
+    output_band = output_dataset.GetRasterBand(band_id)
+    nodata_value = input_band.GetNoDataValue()
+    output_band.SetNoDataValue(nodata_value)
```

Comment:
It isn't always the case that the output band will share a nodata value with the input. Maybe we can add an optional parameter to set a value different from the input?